src/dune_lang/value.mli

```diff
@@ -34,6 +34,33 @@ module L : sig
   val paths : Path.t list -> t list
   val deps_only : t list -> Path.t list
   val dirs : Path.t list -> t list
+
+  (** Concatenate values to a string, separating each value by a space. *)
   val concat : t list -> dir:Path.t -> string
+
   val to_strings : t list -> dir:Path.t -> string list
 end
+
+module Deferred_concat : sig
```

Comment:
Given that deferring can happen also for other reason, maybe `Deferred` is a better name?

Reply:
How about the name `Builder`? It's quite similar to a string builder from Java.

Reply:
I've gone with `Deferred` for now. @rgrinberg are you worried that people might think it's got something to do with `Async`?

Reply:
This type could just as well be implemented as `type t = Path.t -> Value.t`. It lets you commit to doing operations on lists of values but the operations don't occur until a path is provided so stringification of paths in values can occur relative to the given path. I think it's apt to think of this as deferring those operations even though internally it's implemented as a list of values to concatenate.

Reply:
The type `type t = Path.t -> Value.t` could be used implement far more than the behavior in this module. That's the main problem with this name - it's too general for the relatively specific functionality it represents. In any case, you can leave it as `Deferred`. It's only needed for the package management rules, no use to worry too much about it

Reply:
Would the original name `Deferred_concat` be more suitable in your opinion?